[CI] ReindexDataStreamTransportActionIT testNonExistentDataStream failing

**Build Scans:**
- [elasticsearch-periodic #5318 / release-tests](https://gradle-enterprise.elastic.co/s/fjc2j243jmpfy)
- [elasticsearch-periodic #5312 / release-tests](https://gradle-enterprise.elastic.co/s/2mzay4trt6q4u)
- [elasticsearch-periodic #5306 / release-tests](https://gradle-enterprise.elastic.co/s/2y4mhgrsg3px4)
- [elasticsearch-periodic #5300 / release-tests](https://gradle-enterprise.elastic.co/s/2eseqri2mw7eq)
- [elasticsearch-periodic #5294 / release-tests](https://gradle-enterprise.elastic.co/s/vksfgjzmr6an2)
- [elasticsearch-periodic #5288 / release-tests](https://gradle-enterprise.elastic.co/s/ncnflto2nblro)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:migrate:internalClusterTest" --tests "org.elasticsearch.xpack.migrate.action.ReindexDataStreamTransportActionIT.testNonExistentDataStream" -Dtests.seed=310019ECCC8EFF65 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=ksb-Latn-TZ -Dtests.timezone=America/Adak -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.migrate.action.ReindexDataStreamTransportActionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testNonExistentDataStream),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: unexpected exception type thrown; expected:<org.elasticsearch.ResourceNotFoundException> but was:<java.lang.IllegalStateException>
```

**Issue Reasons:**
- [main] 6 consecutive failures in step release-tests
- [main] 6 failures in test testNonExistentDataStream (2.0% fail rate in 299 executions)
- [main] 6 failures in step release-tests (100.0% fail rate in 6 executions)
- [main] 6 failures in pipeline elasticsearch-periodic (100.0% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -384,12 +384,6 @@ tests:
 - class: org.elasticsearch.upgrades.HealthNodeUpgradeIT
   method: testHealthNode {upgradedNodes=2}
   issue: https://github.com/elastic/elasticsearch/issues/118158
-- class: org.elasticsearch.xpack.migrate.action.ReindexDataStreamTransportActionIT
-  method: testNonExistentDataStream
-  issue: https://github.com/elastic/elasticsearch/issues/118275
-- class: org.elasticsearch.xpack.migrate.action.ReindexDataStreamTransportActionIT
-  method: testAlreadyUpToDateDataStream
-  issue: https://github.com/elastic/elasticsearch/issues/118276
 - class: org.elasticsearch.xpack.inference.DefaultEndPointsIT
   method: testInferDeploysDefaultRerank
   issue: https://github.com/elastic/elasticsearch/issues/118184

